tests/e2e/concurrent_write/write.py

```diff
@@ -0,0 +1,103 @@
+import sys
+import threading
+import time
+import typing
+
+import mgclient
+import pytest
+
+
+def execute_and_fetch_all(cursor: mgclient.Cursor, query: str, params: dict = {}) -> typing.List[tuple]:
+    cursor.execute(query, params)
+    return cursor.fetchall()
+
+
+class AtomicInteger:
+    def __init__(self, value=0):
+        self._value = int(value)
+        self._lock = threading.Lock()
+
+    def inc(self, d=1):
+        with self._lock:
+            self._value += int(d)
+            return self._value
+
+    def dec(self, d=1):
+        return self.inc(-d)
+
+    @property
+    def value(self):
+        with self._lock:
+            return self._value
+
+    @value.setter
+    def value(self, v):
+        with self._lock:
+            self._value = int(v)
+            return self._value
+
+
+def sleep_until(wanted_cnt):
+    while cnt.value != wanted_cnt:
+        time.sleep(0.1)
+
+
+def client_success():
+    connection = mgclient.connect(host="localhost", port=7687)
+    connection.autocommit = False
+    cursor = connection.cursor()
+
+    execute_and_fetch_all(cursor, "MATCH (n1:N1) DELETE n1;")
+    cnt.inc()  # 1
+    sleep_until(2)
```

Comment:
Sleep will not give you deterministic behavior. So sometimes this test could be flaky.
You can rely on `cnt` value or use conditional variable to control the behavior.

Reply:
Oops realized how does `sleep_until` work. Good job! 👍